src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -684,13 +681,27 @@ else if (selectorType.equals(targetType) ||
 
     // TypePairs should be in sync with the corresponding record in Lower
     record TypePairs(Class<?> from, Class<?> to) {
+
+        private static final Map<TypePairs, String> typePairToName = initialize();
```

Comment:
If `TypePairs.typePairToName` is never modified after initialisation, then it should probably be made immutable:
```suggestion
        private static final Map<TypePairs, String> typePairToName = Map.copyOf(initialize());
```

Reply:
If you really think about it, the `initialize` method itself is somewhat problematic, as it's initializing with byte/short/char on the left, all of which are already converted to int in the of() factory. This should be done in a separate issue.

Reply:
Yes, the "redirected" mappings can simply be removed in the current implementation. 

Using `Map.copyOf` should be ok to nail down the read-only intent.